### PR TITLE
Issue #172 - handle transient scheduler connection issues

### PR DESCRIPTION
**Pull request by shamsimam**

### CODE REVIEW

File: waiter/src/waiter/scheduler.clj [hunk 1]
```diff
@@ -119,7 +119,10 @@
                                     (ss/try+
                                       {:success (f)}
                                       (catch #(contains? retry-status-codes (:status %)) e
-                                        (log/warn (str "Scheduler unavailable (Error code:" (:status e) ").") msg)
+                                        (log/warn (str "scheduler unavailable (error code:" (:status e) ").") msg)
+                                        (ss/throw+ e))
+                                      (catch ConnectException e
```
Comment (sradack) [inline waiter/src/waiter/scheduler.clj]: Should we handle TimeoutException?  Why shouldn't we handle any exception at all in the same manner?

**Comment by sradack** [comment]
What are you planning to check inside `/state` to ensure test if Waiter can talk to Marathon?

File: waiter/src/waiter/scheduler.clj (hunk 1 above)
Comment (shamsimam) [reply waiter/src/waiter/scheduler.clj]: Other exceptions might not be transient.

**Comment by shamsimam** [comment]
> What are you planning to check inside /state to ensure test if Waiter can talk to Marathon?

This change can still cause the router to fail fatally if the connection error is not transient and persists beyond the retry period.

**Comment by shamsimam** [comment]
I am working on a separate PR that ensures the router keeps running even if the scheduler connections fail and exposing a waiter codahale metric gauge that specifies the last time a router heard from the scheduler.

File: waiter/src/waiter/scheduler.clj (hunk 1 above)
Comment (sradack) [reply waiter/src/waiter/scheduler.clj]: Can we find out what exception occurs on timeouts and handle that as well?

Comment (shamsimam) [reply waiter/src/waiter/scheduler.clj]: Okay

Comment (shamsimam) [reply waiter/src/waiter/scheduler.clj]: Done.

**Comment by shamsimam** [comment]
RTM

**Pull request merged**